src/Utilities/test/runtests.jl

```diff
@@ -85,3 +86,74 @@ using LinearAlgebra
   @test liquid_ice_pottemp.([T, T], .1*[MSLP, MSLP], [0, 1], [0, 0], [0, 0]) ≈
     T .* 10 .^[R_d/cp_d, R_v/cp_v]
 end
+
+@testset "RootSolvers" begin
+  T = (Float32, Float64)
+  for t in T
+    x_star2 = t(10000.0)
+    f_close(x) = x^2 - x_star2
+    f_args(x, y) = x^2 - y
+    x_star = sqrt(x_star2)
+    x_0 = t(0.0)
+    x_1 = t(1.0)
+    args = Tuple(x_star2)
+    tol_abs = t(1.0e-3)
+    iter_max = 100
+    x_root, converged = RootSolvers.find_zero(f_args,
+                                              x_0, x_1,
+                                              args,
+                                              IterParams(tol_abs, iter_max),
+                                              SecantMethod()
+                                              )
+    @test x_root ≈ x_star
+    x_root, converged = RootSolvers.find_zero(f_close,
+                                              x_0, x_1,
+                                              IterParams(tol_abs, iter_max),
+                                              SecantMethod()
+                                              )
+    @test x_root ≈ x_star
+  end
+end
+
+const HAVE_CUDA = try
+  using CUDAdrv
+  using CUDAnative
+  using CuArrays
+  true
+catch
+  false
+end
+@testset "CUDA RootSolvers" begin
+  if HAVE_CUDA
+    x_ca = cu(rand(5, 5))
+    x_ca_0 = x_ca
+    x_ca_1 = x_ca.+2
+    t = typeof(x_ca[1])
+    x_star2 = t(10000.0)
```

Comment:
The important thing for the GPU compiler is that:

1. `x_star2` is constant for the scope of the closure, so you can't modify it outside the closure.
2. That it is an immutable & pointer free struct, e.g. doesn't contain a reference to data residing on the GPU, sadly this includes `CuArray`.
3. If it is a global variable, the variable needs to be marked constant

Sometimes it is helpful to use a let block to ensure 1. and 3.

```
let x_star2=x_star2
   f_close(x) = x^2 - x_star2
end
```